File: SearchingAndSorting/SequentialSearch/JS/SequentialSearch.js
//Sequential Search

function seq_search(arr, ele) {
	/*
	General Sequential Search. Works on Unordered lists.
	*/
	for (var i = 0; i < arr.length; i++) {
		if (arr[i] === ele) {
			return true;
		}
	}
	return false;
}


arr = [1, 9, 2, 8, 3, 4, 7, 5, 6];
console.log(seq_search(arr, 1));
console.log(seq_search(arr, 10));

// #Ordered List
function seq_search2(arr, ele) {
	/*
	General Sequential Search. Works on Unordered lists.
	*/
	for (var i = 0; i < arr.length; i++) {
		if (arr[i] === ele) {
			return true;
		} else if (arr[i] > ele) {
			console.log(i, arr[i],"so stop here");
			return false
		}
	}
	return false;
}

arr = [1, 9, 2, 8, 3, 4, 7, 5, 6];
arr.sort();
console.log(seq_search2(arr, 2.3));
console.log(seq_search2(arr, 9));

File: SearchingAndSorting/SequentialSearch/python/SequentialSearch.py
#Sequential Search

def seq_search(arr,ele):
  """
  General Sequential Search. Works on Unordered lists.
  """
  # Start at position 0
  pos = 0
  # Target becomes true if ele is in the list
  found = False

  # go until end of list
  while pos < len(arr) and not found:
        
  # If match
    if arr[pos] == ele:
      found = True
          
      # Else move one down
    else:
      pos  = pos+1

  return found

arr = [1,9,2,8,3,4,7,5,6]
print seq_search(arr,1)
print seq_search(arr,10)    

#Ordered List
def seq_search2(arr,ele):
  """
  Sequential search for an Ordered list
  """
  # Start at position 0
  pos = 0
  # Target becomes true if ele is in the list
  found = False

  # Stop marker
  stopped = False

  # go until end of list
  while pos < len(arr) and not found and not stopped:
        
    # If match
    if arr[pos] == ele:
      found = True
          
      # Else move one down
    else:
      # Check if element is greater
      if arr[pos] > ele:
        # print(pos, arr[pos])
        stopped = True
                
      # Otherwise move on
      else:
        pos  = pos+1

  return found

arr = [1,9,2,8,3,4,7,5,6]
arr.sort()
print seq_search2(arr,6.5)
print seq_search2(arr,10)  